src/EduManagementLab.Web/Pages/Users/Details.cshtml.cs

```diff
@@ -14,22 +14,57 @@ namespace EduManagementLab.Web.Pages.Users
 {
     public class DetailsModel : PageModel
     {
-        private readonly UserService _userService;
+        private readonly UserService _userService; 
+        private readonly CourseService _courseService;
 
-        public DetailsModel(UserService userService)
+        public DetailsModel(UserService userService, CourseService courseService)
         {
             _userService = userService;
+            _courseService = courseService;
         }
 
+        public List<Course.Membership>? CourseList { get; set; }
         public User User { get; set; }
+        public Course Course { get; set; }
+        public CourseItem CourseListItem { get; set; }
+        public class CourseItem
+        {
+            public Guid courseId { get; set; }
+            public string Code { get; set; }
+            public string Name { get; set; }
+            public DateTime StartDate { get; set; }
+            public DateTime EndDate { get; set; }
+        }
 
-        public async Task<IActionResult> OnGetAsync(Guid id)
+        public async Task<IActionResult> OnGetAsync(Guid userId)
         {
             try
             {
-                User = _userService.GetUser(id);
+                User = _userService.GetUser(userId);
+                CourseList = _courseService.GetUserCourses(User.Id).ToList();
```

Comment:
I get a compile error.
'CourseService' does not contain a definition for 'GetUserCourses'

Reply:
ah right, the GetUserCourses method belongs to another issue. Should i just merge them together here?

Reply:
You could solve two issues in the same pull request if its suitable to do them together, but then you should rename it.
Or you do one first and then the other, but then you have to do them in the right order if they are dependent on each other.